### PR TITLE
feat(amplify): App, Branch and Domain

### PR DESCRIPTION
**Pull request by jogold**
Add L2 constructs for AWS Amplify Console: App, Branch and Domain.

Apps and branches can be password protected with a Secrets Manager generated password.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
@shivlaks @MrArnoldPalmer any update here? anything I can do to help?

**Comment by MrArnoldPalmer** [comment]
@shivlaks and I have been talking about this and playing with it a bit. Overall It feels pretty good but we both want to familiarize ourselves with the service overall a bit more. We just recently started this so expect some feedback sometime this week.

**Comment by jogold** [comment]
Got it, thanks for the feedback.

File: packages/@aws-cdk/aws-amplify/lib/app.ts [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+  readonly oauthToken?: SecretValue;
+
+  /**
+   * The repository for the application. Must use the `HTTPS` protocol.
+   *
+   * @example https://github.com/aws/aws-cdk
+   *
+   * @default - not connected to a repository
+   */
+  readonly repository?: string;
```
[the remark this answers is not in the document]
Comment (jogold) [reply packages/@aws-cdk/aws-amplify/lib/app.ts]: Not tested but yes it should be the case.

> Q: What Git source code providers does AWS Amplify Console support?
You can connect private and public repositories from GitHub, BitBucket, GitLab, and AWS CodeCommit.

https://aws.amazon.com/amplify/console/faqs/

**Pull request merged**